inc/commoninjectionlib.class.php

```diff
@@ -1793,9 +1793,14 @@ private function dataAlreadyInDB($injectionClass, $itemtype) {
                if ($itemtype == $this->primary_type) {
                   foreach ($this->mandatory_fields[$itemtype] as $field => $is_mandatory) {
                      if ($is_mandatory) {
-                        $option = self::findSearchOption($searchOptions, $field);
-                        $where .= " AND `" . $field . "`='".
-                          $this->getValueByItemtypeAndName($itemtype, $field) . "'";
+                        if ($item instanceof User && $field == "useremails_id") {
+                           $email = addslashes($this->getValueByItemtypeAndName($itemtype, $field));
```

Comment:
`$DB->escape()` should be used instead of `addslashes()`

https://www.php.net/manual/en/function.addslashes.php
> The addslashes() is sometimes incorrectly used to try to prevent [SQL Injection](https://www.php.net/manual/en/security.database.sql-injection.php). Instead, database-specific escaping functions and/or prepared statements should be used.The addslashes() is sometimes incorrectly used to try to prevent [SQL Injection](https://www.php.net/manual/en/security.database.sql-injection.php). Instead, database-specific escaping functions and/or prepared statements should be used.

```suggestion
                           $email = $DB->escape($this->getValueByItemtypeAndName($itemtype, $field));
```